### PR TITLE
[1LP][RFR] Host Collections

### PR DESCRIPTION
**Pull request by otsuman**
{{pytest: cfme/tests/ -k 'test_delete_host_appear_after_refresh  or test_host_noquads or test_can_open_advanced_search or test_run_datastore_analysis or  test_run_host_analysis or test_host_drift_analysis or test_host_provisioning  or test_host_good_creds or set_host_credentials or test_set_default_host_filter or test_infra_host_event' }}

PRT failures  not related to changes

### CODE REVIEW

**Comment by dajoRH** [comment]
Lint report for commit 06d2f5d24ab85d9b50f9b09e9cb0d301fdb3075c:

`cfme/tests/infrastructure/test_advanced_search_host.py`:
- :red_circle: Line 7:1: **F401** *'cfme.infrastructure.host' imported but unused*

`cfme/tests/perf/workloads/test_smartstate_analysis.py`:
- :bangbang: Line 72:22: **E225** *missing whitespace around operator*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit 7def6a04f7bd9121d1beed9b147f280b732c2294:

`cfme/infrastructure/host.py`:
- :red_circle: Line 38:14: **F821** *undefined name 'Host'*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit c8cffbf62fb12bf4a83ed702f761178043223e2b:

`cfme/infrastructure/host.py`:
- :heavy_exclamation_mark: Line 387:1: **W293** *blank line contains whitespace*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit ecbd6aab097544ce942bcf5a945edb860f63db33:

`cfme/infrastructure/host.py`:
- :bangbang: Line 479:17: **E126** *continuation line over-indented for hanging indent*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/infrastructure/host.py [hunk 1]
```diff
@@ -403,8 +364,154 @@ def _select_rows(indexes):
...
+                                provider=provider)
+        view = host.create_view(HostsView)
+        assert view.is_displayed
+        view.flash.assert_success_message(flash_message)
+        return host
+
+    def all(self, provider):
+        """returning all Object Store Objects"""
+        view = navigate_to(self, 'All')
+        view.toolbar.view_selector.select("Grid View")
```
Comment (quarckster) [inline cfme/infrastructure/host.py]: **REQUIRED**
It's not necessary, `including_entities` attribute of `HostView` represents all types of entities.

File: cfme/infrastructure/host.py [hunk 2]
```diff
@@ -403,8 +364,154 @@ def _select_rows(indexes):
...
+        hosts = list(hosts)
+        checked_hosts = list()
+        view = navigate_to(self, 'All')
+
+        for host in hosts:
+            try:
+                view.entities.get_entity(by_name=host.name, surf_pages=True).check()
+                checked_hosts.append(host)
+            except ItemNotFound:
+                raise ValueError('Could not find host {} in the UI'.format(host.name))
```
Comment (quarckster) [inline cfme/infrastructure/host.py]: **REQUIRED**
```python
hosts = list(hosts)
checked_hosts = list()
view = navigate_to(self, 'All')
for host in hosts:
    try:
        view.entities.get_entity(by_name=host.name, surf_pages=True).check()
        checked_hosts.append(host)
    except ItemNotFound:
        raise ValueError('Could not find host {} in the UI'.format(host.name))
```
Made a separate method for that in order to not repeat the code.

Comment (otsuman) [reply cfme/infrastructure/host.py]: done

Comment (quarckster) [reply cfme/infrastructure/host.py]: I think `ItemNotFound` should be raised here

File: cfme/infrastructure/host.py [hunk 3]
```diff
@@ -403,8 +364,127 @@ def _select_rows(indexes):
...
+                                ipmi_address=ipmi_address, mac_address=mac_address,
+                                credentials=credentials, ipmi_credentials=ipmi_credentials,
+                                provider=provider)
+        view = host.create_view(HostsView)
+        assert view.is_displayed
+        view.flash.assert_success_message(flash_message)
+        return host
+
+    def all(self, provider):
+        """returning all Object Store Objects"""
```
Comment (quarckster) [inline cfme/infrastructure/host.py]: **REQUIRED**
Copy paste error here 😆

File: cfme/infrastructure/host.py [hunk 4]
```diff
@@ -403,8 +364,127 @@ def _select_rows(indexes):
...
+        view = self.check_hosts(hosts)
+        view.toolbar.configuration.item_select('Perform SmartState Analysis', handle_alert=True)
+        for host in hosts:
+            view.flash.assert_success_message(
+                '"{}": Analysis successfully initiated'.format(host.name))
+
+    def delete(self, *hosts):
+        """Deletes this host from CFME."""
+        view = self.check_hosts(hosts)
+
```
Comment (quarckster) [inline cfme/infrastructure/host.py]: **REQUIRED**
Please remove these extra blank lines.

File: cfme/tests/configure/test_visual_infra.py [hunk 5]
```diff
@@ -10,7 +10,7 @@
 from cfme.infrastructure import virtual_machines as vms  # NOQA
 from cfme.infrastructure.provider import InfraProvider
 from cfme.utils.appliance.implementations.ui import navigate_to
-from cfme.infrastructure.host import Host
+
```
Comment (quarckster) [inline cfme/tests/configure/test_visual_infra.py]: **REQURIED**
Please remove this blank line

File: cfme/tests/infrastructure/test_timelines.py [hunk 6]
```diff
@@ -17,6 +15,7 @@
 from cfme.utils.log import logger
 from cfme.utils.providers import ProviderFilter
 from cfme.utils.wait import wait_for
+from fixtures.provider import setup_one_or_skip
```
Comment (quarckster) [inline cfme/tests/infrastructure/test_timelines.py]: Why did you import it?

File: cfme/tests/openstack/cloud/test_instances.py [hunk 7]
```diff
@@ -172,18 +171,18 @@ def test_delete_instance(new_instance):
...
+    host_collection = appliance.collections.hosts
     # Match hypervisors by IP with count of running VMs
     hvisors = {hv.host_ip: hv.running_vms for hv in provider.mgmt.api.hypervisors.list()}
 
     # Skip non-compute nodes
-    host_names = [e for e in view.entities.all_entity_names if 'Compute' in e]
-    for host_name in host_names:
-        host = Host(host_name, provider=provider.infra_provider)
+    hosts = [host for host in host_collection.all(provider) if 'Compute' in host.name]
+    assert len(hosts) > 0
```
Comment (quarckster) [inline cfme/tests/openstack/cloud/test_instances.py]: **REQURIED**
Here and further
```python
assert hosts
```

File: cfme/tests/infrastructure/test_timelines.py (hunk 6 above)
Comment (otsuman) [reply cfme/tests/infrastructure/test_timelines.py]: Just put this import in order

**Comment by psav** [comment]
Pending PRT I'm very happy for this to go in

**Comment by otsuman** [comment]
@mfalesni It is not related to current PR. As I can see `default_host_filter_btn` was removed in hosts widgetastic conversion 
https://github.com/ManageIQ/integration_tests/commit/7822d814f670227da65112b7d0c43cd707075414#diff-a22636ce4b4d6b0fd3fe95693280ccdbL91

**Comment by mfalesni** [comment]
@otsuman Good, I thought so. So I will then merge this as the errors don't look like conversion artifacts.

**Pull request merged**